### ISSUE TITLE
[SYCL][CTS] Non conformant ```get_pointer()``` for ranged accessors

### ISSUE THREAD

**Issue by hdelan** [intel/llvm#5033]
The SYCL spec states that the member function ```get_pointer()```:

```
Returns a pointer to the start of this accessor’s memory.

For a buffer accessor this is a pointer to the start of the underlying buffer, even if this is a 
ranged accessor whose range does not start at the beginning of the buffer.
```
 (Section 4.7.6.12. Common members for buffer and local accessors)

This is not the behaviour seen in DPC++. In DPC++, ```get_pointer()``` will return a pointer to the start of the buffer + the offset that the accessor is constructed with.

```
#include <CL/sycl.hpp>

int main() {
  std::vector<float> a{5};
  auto bA = sycl::buffer{a};
  auto q = sycl::queue{sycl::default_selector{}};

  q.submit([&](sycl::handler & cgh) {
      constexpr auto offset1 {0}, offset2 {1};
     
      // both accessors access the same buffer with different offsets
      auto acc1 = sycl::accessor{bA, cgh, sycl::range<1>{5}, sycl::range<1>{offset1}}; 
      auto acc2 = sycl::accessor{bA, cgh, sycl::range<1>{4}, sycl::range<1>{offset2}}; 

      cgh.single_task([=](){
          if (acc1.get_pointer() == acc2.get_pointer()) {
            printf("get_pointer returns the same pointers from different ranged accessors. "
                                         "This is the behavior described in the spec.\n");           
          } else {
            printf("get_pointer returns different pointers from different ranged accessors. " 
                                         "You must be using DPC++.\n");
          }
      });
  });
  q.wait();
}
```

This is preventing DPC++ compatibility with many SYCL libraries not built with DPC++.

The change may require significant changes in existing DPC++ codebases.

**Comment by bader** [comment]
I think https://github.com/intel/llvm/pull/4687 addresses this issue.

**Comment by hdelan** [comment]
Yes it does thank you. What is the status of #4687 ?

**Comment by bader** [comment]
It looks like, it's waiting for reviewers. Is there any urgency with fixing this issue
?

**Comment by hdelan** [comment]
I think there is some urgency. It is preventing SYCL-DNN, SYCL-BLAS and other libraries from being used with DPC++. It may be causing other problems with SYCL-CTS as well, but I am not sure.

**Comment by hdelan** [comment]
Resolved by #4687
